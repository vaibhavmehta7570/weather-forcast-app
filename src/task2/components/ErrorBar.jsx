import React from 'react'

const ErrorBar = ({ message }) => {
  return (
    <section className="error">
      <figure className="error__figure">¯\_(ツ)_/¯</figure>
      <label className="error__message">{message}</label>
    </section>
  )
}

export default ErrorBar
