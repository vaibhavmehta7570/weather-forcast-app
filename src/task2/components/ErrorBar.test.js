import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import ErrorBar from './ErrorBar'
import renderer from 'react-test-renderer'
configure({ adapter: new Adapter() })

describe('Should test WeatherApp.jsx', () => {
  const wrapperErrorBar = shallow(<ErrorBar message={'Mock error message'} />)

  test('should render correctly Counter component', () => {
    const appRenderResponse = renderer.create(<ErrorBar />).toJSON()
    expect(appRenderResponse).toMatchSnapshot()
  })

  test('should simulate errorbar displaying error prop', () => {
    const value = wrapperErrorBar.find('.error__message').text()
    expect(value).toBe('Mock error message')
  })
})
