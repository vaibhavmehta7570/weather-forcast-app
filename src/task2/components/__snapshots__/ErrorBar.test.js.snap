// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test WeatherApp.jsx should render correctly Counter component 1`] = `
<section
  className="error"
>
  <figure
    className="error__figure"
  >
    ¯\\_(ツ)_/¯
  </figure>
  <label
    className="error__message"
  />
</section>
`;
